### PR TITLE
server: add health check service

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

What's Changed:

This can allow client to use the builtin health check of grpc to detect
failure of networking.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 
- No release note

### CODE REVIEW

**Review by MyonKeminta — APPROVED** [review]

File: tests/integrations/server/kv_service.rs [hunk 1]
```diff
@@ -1562,3 +1564,24 @@ fn test_forwarding_reconnect() {
     let resp = client.raw_get_opt(&req, call_opt.clone()).unwrap();
     assert!(!resp.get_region_error().has_store_not_match(), "{:?}", resp);
 }
+
+#[test]
+fn test_health_check() {
```
Comment (Connor1996) [inline tests/integrations/server/kv_service.rs:1569]: should we cover `watch` too?

Comment (BusyJay) [reply tests/integrations/server/kv_service.rs:1569]: This only means to check if changes are made by TiKV. Whether `watch` works should guaranteed by grpcio itself.

**Review by MyonKeminta — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]

**Comment by hi-rustin** [comment]
@ti-chi-bot ?

**Pull request merged**